provider/gce/environ_network.go

```diff
@@ -1,40 +1,213 @@
+				iface.ProviderSubnetId,
+				iface.CIDR,
+				zones,
+			))
+		}
+	}
+	return results, nil
+}
+
+// NetworkInterfaces implements environs.NetworkingEnviron.
+func (e *environ) NetworkInterfaces(instId instance.Id) ([]network.InterfaceInfo, error) {
+	insts, err := e.Instances([]instance.Id{instId})
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	envInst, ok := insts[0].(*environInstance)
+	if !ok {
+		// This shouldn't happen.
+		return nil, errors.Errorf("couldn't extract google instance for %q", instId)
+	}
+	googleInst := envInst.base
+	// According to GCE docs, an instance can only have one nic.
+	// https://cloud.google.com/compute/docs/instances/#instances_and_networks
+	if ifaces := len(googleInst.NetworkInterfaces()); ifaces != 1 {
+		return nil, errors.Errorf("expected 1 network interface on %q, found %d", instId, ifaces)
+	}
+	iface := googleInst.NetworkInterfaces()[0]
+	subnetId := network.Id(iface.Subnetwork)
+	subnets, err := e.Subnets(instance.UnknownId, []network.Id{subnetId})
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	subnet := subnets[0]
+	results := []network.InterfaceInfo{{
+		DeviceIndex: 0,
+		CIDR:        subnet.CIDR,
+		// XXX(xtian): not sure about this - the network interface has
+		// no id in GCE, each machine has exactly one interface, so it
+		// can be identified by the machine's id.
+		ProviderId:        network.Id(instId),
+		ProviderSubnetId:  subnetId,
+		AvailabilityZones: subnet.AvailabilityZones,
+		InterfaceName:     iface.Name,
+		Address:           network.NewScopedAddress(iface.NetworkIP, network.ScopeCloudLocal),
+		InterfaceType:     network.EthernetInterface,
+		Disabled:          false,
+		NoAutoStart:       false,
+		ConfigType:        network.ConfigDHCP,
+	}}
+	return results, nil
+}
+
+// SupportsSpaces implements environs.NetworkingEnviron.
+func (e *environ) SupportsSpaces() (bool, error) {
+	return false, nil
+}
+
+// SupportsSpaceDiscovery implements environs.NetworkingEnviron.
+func (e *environ) SupportsSpaceDiscovery() (bool, error) {
+	return false, nil
+}
+
+// Spaces implements environs.NetworkingEnviron.
+func (e *environ) Spaces() ([]network.SpaceInfo, error) {
+	return nil, errors.NotSupportedf("spaces")
+}
+
+// SupportsContainerAddresses implements environs.NetworkingEnviron.
+func (e *environ) SupportsContainerAddresses() (bool, error) {
+	return false, nil
+}
+
+// AllocateContainerAddresses implements environs.NetworkingEnviron.
+func (e *environ) AllocateContainerAddresses(instance.Id, names.MachineTag, []network.InterfaceInfo) ([]network.InterfaceInfo, error) {
+	return nil, errors.NotSupportedf("container addresses")
+}
+
+// ReleaseContainerAddresses implements environs.NetworkingEnviron.
+func (e *environ) ReleaseContainerAddresses([]network.ProviderInterfaceInfo) error {
+	return errors.NotSupportedf("container addresses")
+}
+
+func makeSubnetInfo(subnetId network.Id, cidr string, zones []string) network.SubnetInfo {
```

Comment:
Yeah, I think you're right - I've done that for default-value fields in `InterfaceInfo`, for example. I'll add those in.